frontend/src/js/controllers/challengeCreate.js

```diff
@@ -7,21 +7,86 @@
         .module('evalai')
         .controller('ChallengeCreateCtrl', ChallengeCreateCtrl);
 
-    ChallengeCreateCtrl.$inject = ['utilities', '$state'];
+    ChallengeCreateCtrl.$inject = ['utilities', 'loaderService', '$rootScope', '$state'];
 
-    function ChallengeCreateCtrl(utilities, $state) {
+    function ChallengeCreateCtrl(utilities, loaderService, $rootScope, $state) {
         var vm = this;
         var userKey = utilities.getData('userKey');
-        var hostTeamId = utilities.getData('challengeHostTeamPk');
+        var hostTeamId = utilities.getData('challengeHostTeamId');
         vm.wrnMsg = {};
         vm.isValid = {};
+        vm.isFormError = false;
+        vm.input_file = null;
+        vm.formError = {};
 
-        // function to create a challenge
-        vm.createChallenge = function() {
-            var parameters = {};
-            parameters.url = 'challenges/challenge_host_team/' + hostTeamId +'/challenge';
-            parameters.method = 'POST';
-            parameters.data = {
+        // start loader
+        vm.startLoader = loaderService.startLoader;
+
+        // stop loader
+        vm.stopLoader = loaderService.stopLoader;
+
+        // function to create a challenge using zip file.
+    vm.challengeCreate = function() {
+            if (hostTeamId) {
+                var fileVal = angular.element(".file-path").val();
+
+                if (fileVal === null || fileVal === "") {
+                    vm.isFormError = true;
+                    vm.formError = "Please upload file!";
+                }
+                if (vm.input_file) {
+                    var parameters = {};
+                    parameters.url = 'challenges/challenge/' + hostTeamId + '/zip/';
+                    parameters.method = 'POST';
+                    var formData = new FormData();
+                    formData.append("zip_configuration", vm.input_file);
+
+                    parameters.data = formData;
+
+                    parameters.token = userKey;
+                    parameters.callback = {
+                        onSuccess: function(response) {
+                            console.log(response);
+                            var status = response.status;
+                            var details =  response.data;
+                            if (status == 201) {
+
+                                angular.forEach(
+                                    angular.element("input[type='file']"),
```

Comment:
`angular.element("input[type=''file"]")`  should be outside the angular for each loop, save it inside the variable.